crates/sdk/src/sys.rs

```diff
@@ -1,30 +1,54 @@
 #![allow(dead_code)]
 
+#[cfg(not(target_arch = "wasm32"))]
 mod types;
 
 pub use types::*;
 
-#[link(wasm_import_module = "env")]
-extern "C" {
-    pub fn panic(loc: Location) -> !;
-    pub fn panic_utf8(msg: Buffer, loc: Location) -> !;
-    // --
-    pub fn register_len(register_id: RegisterId) -> PtrSizedInt;
-    pub fn read_register(register_id: RegisterId, buf: BufferMut) -> Bool;
-    // --
-    pub fn input(register_id: RegisterId);
-    pub fn value_return(value: ValueReturn);
-    pub fn log_utf8(msg: Buffer);
-    pub fn emit(event: Event);
-    // --
-    pub fn storage_read(key: Buffer, register_id: RegisterId) -> Bool;
-    pub fn storage_write(key: Buffer, value: Buffer, register_id: RegisterId) -> Bool;
+macro_rules! wasm_imports {
```

Comment:
let's move the macro definition to the bottom of the file since the methods are the important bit here

it'll complain, since macros must be in scope before their use, that's okay, we can abuse `use`

just add this after the macro definition:

```rust
use wasm_imports;
```